java/oauth-custom-challengehandler/src/main/java/com/esri/arcgisruntime/sample/oauthcustomchallengehandler/MainActivity.java

```diff
@@ -0,0 +1,304 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.oauthcustomchallengehandler;
+
+import java.net.MalformedURLException;
+import java.util.concurrent.CountDownLatch;
+
+import android.app.Dialog;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.ViewGroup;
+import android.webkit.WebSettings;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+import android.widget.FrameLayout;
+import android.widget.Toast;
+
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.portal.Portal;
+import com.esri.arcgisruntime.portal.PortalItem;
+import com.esri.arcgisruntime.security.AuthenticationChallenge;
+import com.esri.arcgisruntime.security.AuthenticationChallengeHandler;
+import com.esri.arcgisruntime.security.AuthenticationChallengeResponse;
+import com.esri.arcgisruntime.security.AuthenticationManager;
+import com.esri.arcgisruntime.security.OAuthConfiguration;
+import com.esri.arcgisruntime.security.OAuthTokenCredential;
+import com.esri.arcgisruntime.security.OAuthTokenCredentialRequest;
+
+public class MainActivity extends AppCompatActivity {
+
+  private MapView mMapView;
+
+  private OAuthConfiguration mOAuthConfiguration = null;
+
+  private Dialog mOAuthLoginDialog;
+
+  /**
+   * Anonymous class to define our mCustomAuthenticationChallengeHandler.
+   */
+  AuthenticationChallengeHandler mCustomAuthenticationChallengeHandler = new AuthenticationChallengeHandler() {
+    /**
+     * Handles the incoming AuthenticationChallenge, returning a response that contains an action and
+     * potentially a parameter with which to carry out the action.
+     *
+     * @param challenge the authentication challenge to handle
+     * @return the response that should be taken
+     */
+    @Override
+    public AuthenticationChallengeResponse handleChallenge(AuthenticationChallenge challenge) {
+
+      AuthenticationChallengeResponse ret = null;
+
+      if (challenge.getType() == AuthenticationChallenge.Type.OAUTH_CREDENTIAL_CHALLENGE) {
+        final OAuthChallengeCountDownLatch oAuthChallengeCountDownLatch = new OAuthChallengeCountDownLatch(1);
+
+        final String portalUrl = challenge.getRemoteResource().getUri();
+
+        // get the OAuth config for the portalUrl associated with the challenge
+        try {
+          mOAuthConfiguration = AuthenticationManager.getOAuthConfiguration(portalUrl);
+        } catch (MalformedURLException e) {
+
+        }
+
+        runOnUiThread(new Runnable() {
```

Comment:
Looks like we need to guard the code from here to dismissing the dialog with a null check:
```
if (mOAuthConfiguration != null) {

   runOnUiThread(new Runnable() {
      ...
   }
   ...
  
   // dismiss the dialog
   mOAuthLoginDialog.dismiss();
}
```

Reply:
added